src/frontend/Preprocessor.mli

```diff
@@ -20,6 +25,8 @@ val update_start_positions : Lexing.position -> unit
 val pop_buffer : unit -> Lexing.lexbuf
 (** Pop the buffer at the top of the include stack *)
 
+val comments : Ast.comment_type Queue.t
```

Comment:
Maybe this should be two functions `val add_comment: Ast.comment_type -> unit` and `val get_comments: unit -> Ast.comment_type list`?